### PR TITLE
Post Content: Update placeholder when editing template

### PR DESCRIPTION
**Pull request by noisysocks**
## What?
Updates the Post Content block's placeholder when viewing it in the context of editing a template.

| Before | After |
| - | - |
| Screenshot 2024-01-29 at 13 44 01 | Screenshot 2024-01-29 at 13 43 37 |

I pulled this out of https://github.com/WordPress/gutenberg/pull/57572. Props to my boy @ramonjd.

## Why?
A common painpoint when using the site editor is that users delete the Post Content block without meaning to. We hope to reduce this by making it look more like placeholder content and less like actual content.

## Testing Instructions
1. Go to Appearance → Editor → Pages and create/edit a page.
2. Edit the template e.g. by selecting Template in the sidebar and clicking _Edit template_.

### CODE REVIEW

**Comment by ramonjd** [comment]
> Props to my boy @ramonjd.

The only thing of value I contributed were probably the updates to the E2E tests:

https://github.com/WordPress/gutenberg/pull/57572/files#diff-b02f5031d46bffd4e323652aa470e4d147dec8dfe1858edd03404ad07279f6d5R127

Which, I think, this PR needs.

Most of it was plagiarized from @SaxonF 😆 , whom I'll let have the final say 🙇🏻 

Otherwise it LGTM 

Screenshot 2024-01-29 at 4 37 00 pm

Thanks!

**Review by ramonjd — APPROVED** [review]
LGTM (pending design 👍🏻)

**Comment by noisysocks** [comment]
Thanks @ramonjd. I think @SaxonF is still ruminating on the design.

**Comment by jasmussen** [comment]
I'm not a believer in this one. I regret to be the wet blanket here, but this seems an arbitrary placeholder to update, and I still don't think this will make a meaningful difference to the otherwise very legitimate issue.

I think we need to look at a proper long term solution for 6.6, one that does not rely on outlines or placeholders or other noise that I worry will become vestigial and feel like bandaids we have to remove in the future. Instead we should focus on smaller improvements, like #58169, or the idea that _clicking anywhere_ when the template and footer are locked, sets focus in the nearest editable field.

**Pull request closed without merging**
